lib/addDefaults.js

```diff
@@ -190,6 +191,59 @@ function lightDefaults(gltf) {
     }
 }
 
+function checkIfFloatVec4(value) {
+    if (defined(value) && value.length === 4) {
+        if (typeof(value[0]) === 'number' &&
+            typeof(value[1]) === 'number' &&
+            typeof(value[2]) === 'number' &&
+            typeof(value[3]) === 'number') return value;
+    }
+    else return [0, 0, 0, 1];
+}
+
+function guessExtensionTechnique(material) {
+    // Replaces the material technique, extension, and values with a close equivalent from KHR_materials_common.
+    // If no equivalent can be found, defaults to blinn for specular materials and lambert otherwise
+    var technique = 'LAMBERT';
+    var useBlinn = false;
+    var values = defined(material.values) ? material.values : {};
+
+    var ambient = checkIfFloatVec4(values.ambient);
+    var diffuse = typeof(values.diffuse) === 'string' ? values.diffuse : checkIfFloatVec4(values.diffuse);
+    var doubleSided = typeof(values.doubleSided) === 'boolean' ? values.doubleSided : false;
+    var emission = typeof(values.emission) === 'string' ? values.emission : checkIfFloatVec4(values.emission);
+    var specular = typeof(values.specular) === 'string' ? values.specular : checkIfFloatVec4(values.specular);
+    var shininess = typeof(values.shininess) === 'number' ? values.shininess : 0.0;
```

Comment:
We're keeping the `typeof` checks to make sure something like `shininess` as a texture map in a more customized gltf doesn't make its way into the `KHR_materials_common`.